src/targets/registry.ts

```diff
@@ -367,16 +370,18 @@ export class RegistryTarget extends BaseTarget {
     );
 
     // Commit
-    await git.checkout('master');
     await git.add(['.']);
     await git.commit(`craft: release "${canonicalName}", version "${version}"`);
 
+    // Ensure we are still up to date with upstream
+    await git.pull('origin', 'master', { rebase: true });
```

Comment:
There's still a chance for a race condition. What if you put this in a bounded loop?

Reply:
I think, at our scale it is very unlikely that we'll hit the condition you mention and a bounded loop is still a band-aid fix. When that time comes, we should look into proper queues, not more band aids.